### ISSUE TITLE
Different view of the scheduler

### ISSUE THREAD

**Issue by DamonHD** [davidwernhart/AlDente#980]
**Is your feature request related to a problem? Please describe.**
I think a different view of scheduling may work better for many cases.

**Describe the solution you'd like**
Rather than "do this at this time" and "start tasks at next opportunity" and "stop at this time" etc which is a bit complex and fragile (though is generally working for me), it may be better to have a more declarative "between time X and time Y, while conditions A, B, C are true, do Things1 else Things2 in this time".  So a simple one for me would be "between 4pm and 7pm on weekdays [unconditionally in this case] halve the charge threshold and go into low power mode".  In future some of those conditions could be external, eg poll X for grid intensity.

**Describe alternatives you've considered**
Stay with things the way they are or script or ...

**Comment by MatthiasKerbl** [comment]
Hi @DamonHD,

Thank you for the idea. In general, we want to keep the Schedule feature simple and therefore set it up in a way where different features of AlDente are triggered automatically. For more complex tasks and automations, it is probably better to use the Apple Shortcuts integration of AlDente due to tighter integration with macOS.

Best,
Matthias

**Comment by DamonHD** [comment]
I totally agree with keeping it simple, but IMHO declarative "what I'd like to be happening at this time" stuff is more robust and ultimately more simple that "do this thing now".  That's likely especially true for people who aren't programmers.

**Comment by MatthiasKerbl** [comment]
You are right that for example, a weekly calendar view of tasks might be more clear. However, it is also more difficult to implement and not for all features straight forward, especially if they are not weekly.
How would this different implementation of the schedule feature help with situations when the MacBook is asleep or turned off?

**Comment by DamonHD** [comment]
Hi,

It is difficult to find simple solutions that work for all or even most cases!  Thank you for continuing to consider this.

I suspect that most cyclic activities are daily or weekly or monthly.  But in any case you could lean on the calendaring functionality built into the Mac and have AlDente be an extra calendar, or at least use the iCal spec and a calendar-like view.

This would much more robust when the Mac restarts: just look in the calendar to see what state AlDente should now be in.  (Or state combination where there are overlapping conditions.)

For example I have my charge limit drop from 80% to 40% ~4pm to ~8pm and set lowpowermode, and within ~5pm to ~7pm I additionally switch to discharge mode.  If the Mac awoke at 18:50 it would know to go into lowpowermode, 40% charge limit AND discharge mode straight away.

Rgds

Damon
